### PR TITLE
Increase reusability

### PR DESCRIPTION
**Pull request by Kagee**
AUTOCONF_GPG and AUTOCONF_ADMIN_KEY to turn on(default, true) or off features. Print error to logs if  AUTOCONF_ADMIN_KEY affects automation. Optional MISP_EMAIL for setting MISP.email, default ADMIN_EMAIL. Optional MISP_CONTACT for setting MISP.contact, default ADMIN_EMAIL. Configuring MISP.email moved out of init_user, as it is not really user init.

Minor linting.

### CODE REVIEW

File: server/files/configure_misp.sh [hunk 1]
```diff
@@ -227,7 +238,9 @@ echo "MISP | Resolve critical issues ..." && apply_critical_fixes
 
 echo "MISP | Resolve non-critical issues ..." && apply_optional_fixes
 
-echo "MISP | Create sync servers ..." && create_sync_servers
+[ -z "$SYNCSERVERS" ] && echo "MISP | No sync servers to create ..."
+[ -n "$SYNCSERVERS" ] && [ -n "$ADMIN_KEY" ] && echo "MISP | Create sync servers ..." && create_sync_servers
+[ -n "$SYNCSERVERS" ] && [ -z "$ADMIN_KEY" ] && echo "MISP | ERROR: Can not create sync servers, admin key not avaliable ..." 1>&2
```
Comment (ostefano) [inline server/files/configure_misp.sh:243]: Why the 1>&2?

Comment (Kagee) [reply server/files/configure_misp.sh:243]: It's an error. Errors goes to stderr :)

Comment (ostefano) [reply server/files/configure_misp.sh:243]: But it's not if this is a valid combination of options, right?

File: server/files/configure_misp.sh [hunk 2]
```diff
@@ -6,12 +6,16 @@ source /rest_client.sh
...
+[ -z "$AUTOCONF_GPG" ] && AUTOCONF_GPG="true"
+[ -z "$AUTOCONF_ADMIN_KEY" ] && AUTOCONF_ADMIN_KEY="true"
+
 
 init_configuration(){
     # Note that we are doing this after enforcing permissions, so we need to use the www-data user for this
     echo "... configuring default settings"
-    sudo -u www-data /var/www/MISP/app/Console/cake Admin setSetting -q "MISP.redis_host" "$REDIS_FQDN"
     sudo -u www-data /var/www/MISP/app/Console/cake Admin setSetting -q "MISP.baseurl" "$HOSTNAME"
+    sudo -u www-data /var/www/MISP/app/Console/cake Admin setSetting -q "MISP.email" "${MISP_EMAIL-$ADMIN_EMAIL}"
```
Comment (ostefano) [inline server/files/configure_misp.sh:17]: MISP_EMAIL never defined?

File: server/files/configure_misp.sh [hunk 3]
```diff
@@ -209,15 +220,15 @@ create_sync_servers() {
...
+    done
 }
 
 
 echo "MISP | Initialize configuration ..." && init_configuration
 
 echo "MISP | Initialize workers ..." && init_workers
 
-echo "MISP | Configure GPG key ..." && configure_gnupg
+[ "$AUTOCONF_GPG" = "true" ] && echo "MISP | Configure GPG key ..." && configure_gnupg
```
Comment (ostefano) [inline server/files/configure_misp.sh:231]: I will move those checks within the function.

File: server/files/configure_misp.sh [hunk 4]
```diff
@@ -146,11 +158,10 @@ apply_critical_fixes() {
 apply_optional_fixes() {
     sudo -u www-data /var/www/MISP/app/Console/cake Admin setSetting -q --force "MISP.welcome_text_top" ""
     sudo -u www-data /var/www/MISP/app/Console/cake Admin setSetting -q --force "MISP.welcome_text_bottom" ""
-    
-    sudo -u www-data /var/www/MISP/app/Console/cake Admin setSetting -q "MISP.contact" "${ADMIN_EMAIL}"
+    sudo -u www-data /var/www/MISP/app/Console/cake Admin setSetting -q "MISP.contact" "${MISP_CONTACT-$ADMIN_EMAIL}"
```
Comment (ostefano) [inline server/files/configure_misp.sh:161]: MISP_CONTACT?

**Comment by Kagee** [comment]
Mostly merged.

**Pull request closed without merging**